contrib_bots/lib/CommuteBot/CommuteBot.py

```
@@ -0,0 +1,219 @@
+                              + 'distancematrix/json',
+                              params=params))
+            rjson = r.json()
+
+            # determines if user has valid inputs
+            try:
+                not_found = (rjson["rows"][0]["elements"][0]["status"]
+                             == "NOT_FOUND")
+                invalid_request = (rjson["status"] == "INVALID_REQUEST")
+                no_result = (rjson["rows"][0]["elements"][0]["status"]
+                             == "ZERO_RESULTS")
+
+                if no_result:
+                    self.send_info(message,
+                                   "Zero results\n"
+                                   + "If stuck, try '@commute help'.", client)
+                    return None
+
+                elif not_found or invalid_request:
+                    raise IndexError
+            except IndexError:
+                self.send_info(message,
+                               "Invalid input, please see instructions."
+                               + "\nIf stuck, try '@commute help'.", client)
+                return None
+
+            # origin and destination strings
+            begin = 'From: ' + rjson["destination_addresses"][0]
+            end = 'To: ' + rjson["origin_addresses"][0]
+            distance = ('Distance: '
+                        + rjson["rows"][0]["elements"][0]["distance"]["text"])
+            duration = ('Duration: '
+                        + rjson["rows"][0]["elements"][0]["duration"]["text"])
+            output = begin + '\n' + end + '\n' + distance
+
+            # determines if fare information is available
+            try:
+                fare = (
+                        'Fare: '
+                        + rjson["rows"][0]["elements"][0]["fare"]["currency"]
+                        + rjson["rows"][0]["elements"][0]["fare"]["text"])
+                output += '\n' + fare
+            except (KeyError, IndexError):
+                print('')
```

Yep. What you are doing is OK too. If you want to ignore the exception you could just have a `pass` statement.